src/discussions/posts/data/slices.js

```diff
@@ -65,10 +43,10 @@ const threadsSlice = createSlice({
     },
     fetchThreadsSuccess: (state, { payload }) => {
       state.status = RequestStatus.SUCCESSFUL;
-      state.topicThreadMap = {};
-      state.threads = {};
-      normaliseThreads(state, payload.results);
-      state.page = payload.pagination.page;
+      state.pages[payload.page - 1] = payload.ids;
```

Comment:
Pages are initialized as an empty array but here it is being changed to object with keys starting from zero. I am unable to comprehend this logic and I see no use of this anywhere. Are you sure this change is part of this PR?

Reply:
@mehaknasir It is not being changed to an object with numeric keys starting from zero. It is still an array. 
You can see this used in the selectors when ids in pages are mapped to items in them.

Reply:
I confirm it again. now its being converted to array. resolving this.